src/ui/public/agg_types/metrics/top.js

```diff
@@ -0,0 +1,68 @@
+import { noop } from 'lodash';
+import MetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import topSortEditor from 'ui/agg_types/controls/top_sort.html';
+
+export default function AggTypeMetricTopProvider(Private) {
+  const MetricAggType = Private(MetricAggTypeProvider);
+
+  return new MetricAggType({
+    name: 'top_hits',
+    title: 'Top',
+    makeLabel: function (aggConfig) {
+      const prefix = aggConfig.params.sortOrder.val === 'desc' ? 'Last' : 'First';
+      return `${prefix} ${aggConfig.params.field.displayName}`;
+    },
+    params: [
+      {
+        name: 'field',
+        filterFieldTypes: function (vis, value) {
+          if (vis.type.name === 'table') {
+            return true;
+          }
+          if (vis.type.name === 'pie') {
+            return value === 'number';
+          }
+          return value === 'number' || value === 'date';
```

Comment:
This is not the best data to show this, but here you go:

![2016-10-11-095509_791x299_scrot](https://cloud.githubusercontent.com/assets/795879/19264553/14a08032-8f99-11e6-902f-bef138873d58.png)

I have some dummy weather data with temperature and dates. I am taking the top date when sorting on the temperature. So for each term bucket (here, the location), I want to show the date the highest temperature was reached.

It doesn't look nice though because the Y axis labels take a lot of space.
My thought when allowing dates was to let as much freedom to a user as possible.